src/Avalonia.Controls/Presenters/ItemVirtualizerSimple.cs

```diff
@@ -200,6 +200,10 @@ public override void ItemsChanged(IEnumerable items, NotifyCollectionChangedEven
                         break;
 
                     case NotifyCollectionChangedAction.Reset:
+                        Owner.ItemContainerGenerator.Clear();
+                        VirtualizingPanel.Children.Clear();
```

Comment:
i think this will dramatically decrease performance, on fast changing lists, as on reset it will recreate all controls instead reuse them, for virtualization turned on i don't think this is a desired outcome, wouldn't it be better to clear/regenerate all containers only when `ItemTemplate ` is changed?

Reply:
Ah yes good point - I'd not considered that!

Reply:
for fast-changing lists you mean a list that rapidly adds and removes items? if so it should not raise the Reset event

Reply:
General guideline for a big change in a list (adding removing a lot of items) is to send a `Reset` event since it is faster. Many frameworks like `DynamicData` are doing this when filtering for example.